go-controller/pkg/libovsdbops/portbinding.go

```diff
@@ -0,0 +1,77 @@
+package libovsdbops
+
+import (
+	"fmt"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/sbdb"
+)
+
+// GetPortBinding looks up a port binding from the cache using portBinding.LogicalPort (which is an indexed column)
+func GetPortBinding(sbClient libovsdbclient.Client, portBinding *sbdb.PortBinding) (*sbdb.PortBinding, error) {
+	found := []*sbdb.PortBinding{}
+	opModel := operationModel{
+		Model:          portBinding,
+		ModelPredicate: func(item *sbdb.PortBinding) bool { return item.LogicalPort == portBinding.LogicalPort },
+		ExistingResult: &found,
+		ErrNotFound:    true,
+		BulkOp:         false,
+	}
+
+	m := newModelClient(sbClient)
+	err := m.Lookup(opModel)
+	if err != nil {
+		return nil, err
+	}
+
+	return found[0], nil
+}
+
+// UpdatePortBindingSetChassis sets the chassis column of the 'portBinding' row so that the OVN thinks that
+// the port binding 'portBinding' is bound on the chassis. Ideally its ovn-controller which claims/binds
+// a port binding. But for a remote chassis, we have to bind it as we created the remote chassis
+// record for the remote zone nodes.
+// TODO (numans) remove this function once OVN supports binding a port binding for a remote
+// chassis.
+func UpdatePortBindingSetChassis(sbClient libovsdbclient.Client, portBinding *sbdb.PortBinding, chassis *sbdb.Chassis) error {
+	remoteChassis, err := GetChassis(sbClient, chassis)
+	if err != nil {
+		return fmt.Errorf("failed to get chassis id %s(%s), error: %v", chassis.Name, chassis.Hostname, err)
+	}
+
+	portBinding, err = GetPortBinding(sbClient, portBinding)
+	if err != nil {
+		return err
+	}
+
+	portBinding.Chassis = &remoteChassis.UUID
+
+	opModel := operationModel{
+		Model:          portBinding,
+		OnModelUpdates: []interface{}{&portBinding.Chassis},
+		ErrNotFound:    true,
+		BulkOp:         false,
+	}
+
+	m := newModelClient(sbClient)
+	_, err = m.CreateOrUpdate(opModel)
+	return err
+}
```

Comment:
Similar comments as I had before.

This seems to be implementing something that should be done at a higher level. Here you would only have a `CreateOrUpdatePortBinding` which receives a `sbdb.PortBinding` with all the data already set.

Also, no need to read the portBinding beforehand.

Reply:
yup ditto ^ @numansiddique we should try to keep the actual logic and libovsdb specific logic separate and not mix them..

Reply:
I agree for the other comments, but here I'd disagree.  I don't think ovnk should create a port binding if it is not present.   It is the job of ovn-northd.

When ovnk creates a remote logical port for a node in the transit switch,  ovn-northd will(and has to) create a port binding in the Southbound db.

Once that is created, ovnk has to bind that remote port binding to the remote chassis (manually).  Ideally this should be done by ovn-northd.  Please see the TODO.  I'll submit a patch soon to add this support in ovn-northd and once that is done we can remove this function.

Also UpdatePortBindingSetChassis is not a higher level function IMO. It is just a helper function to set the chassis column of port binding.

I see a similar function UpdateLogicalSwitchPortSetOptions which updates the logical_switch_port's options column in libovsdops/switch.go

Reply:
> I agree for the other comments, but here I'd disagree. I don't think ovnk should create a port binding if it is not present. It is the job of ovn-northd.

> Also UpdatePortBindingSetChassis is not a higher level function IMO. It is just a helper function to set the chassis column of port binding.

That's fine. If you only want to make available a function to update the chassis, we can have a this `UpdatePortBindingSetChassis` function.

But the whole `GetChassis` and `portBinding.Chassis = &remoteChassis.UUID` can be done outside as a future caller might already have that available and doesn't have a need for this. You are composing operations in a way that a caller might not need them.

On the other side, you have a point here and we do try to hide UUID handling from the caller although it is not something we have always been able to do. So another, maybe more reasonable, option is to check if the passed in `chassis *sbdb.Chassis` has already a UUID set and avoid doing a `GetChassis` in that case. A `GetChassis` wouldn't do any harm but could be avoided.

Reply:
Ack